benchmarks/lib/src/jmh/java/com/amazon/corretto/crypto/provider/benchmarks/AesCbcUpdate.java

```diff
@@ -0,0 +1,58 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.amazon.corretto.crypto.provider.benchmarks;
+
+import java.security.spec.AlgorithmParameterSpec;
+import javax.crypto.spec.IvParameterSpec;
+
+import com.amazon.corretto.crypto.provider.AmazonCorrettoCryptoProvider;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+
+@State(Scope.Benchmark)
+public class AesCbcUpdate extends AesBase {
+  @Param({"128", "256"})
+  public int keyBits;
+
+  @Param({AmazonCorrettoCryptoProvider.PROVIDER_NAME, "BC", "SunJCE"})
+  public String provider;
+
+  @Param({"NoPadding"}) // Other values: { "PKCS5Padding", "ISO10126Padding" }
```

Comment:
I think these would be interesting to add since the memory access patterns are really odd (especially for ISO10126Padding).

Reply:
I did that initially, but ran into problems with over-parameterization. JMH ran out of colors so just blacked-out half the bar graph.

Reply:
Agree that we don't strictly need this, but a 17% reduction relative to NoPadding in throughput is not negligible. If you're willing to humor me, I think it'd be useful to add a parameter for one of the padding schemes.

Reply:
here's what the update benchmark looks like parameterized by padding. i agree that it's worth keeping some visibility on perf differences there. if we notice something, we can de-paramaterize the benchmarks a bit to dig in.

![Screenshot 2025-05-05 at 12 12 38 PM](https://github.com/user-attachments/assets/bbd96c2a-2ea4-48d6-a876-be0ebd18c01d)